index.js

```diff
@@ -690,12 +695,16 @@ class KeyringController extends EventEmitter {
    * On success, returns the resulting keyring instance.
    *
    * @param {Object} serialized - The serialized keyring.
-   * @returns {Promise<Keyring>} The deserialized keyring.
+   * @returns {Promise<Keyring|undefined>} The deserialized keyring or undefined if the keyring type is unsupported.
    */
   async _restoreKeyring(serialized) {
     const { type, data } = serialized;
 
     const Keyring = this.getKeyringClassForType(type);
+    if (!Keyring) {
+      this._unsupportedKeyrings.push(serialized);
+      return undefined;
```

Comment:
hi @jpuri, late question: when we support hard wallets for MV3, will this logic added in this PR still be needed?